js/src/components/tree-select-control/options.js

```diff
@@ -0,0 +1,116 @@
+/**
+ * External dependencies
+ */
+import { CheckboxControl, Flex } from '@wordpress/components';
+import { useState } from '@wordpress/element';
+import { Icon, chevronUp, chevronDown } from '@wordpress/icons';
+import classnames from 'classnames';
+
+/**
+ * This component renders a list of options and its children recursively
+ *
+ * @param {Object} params Component parameters
+ * @param {Option[]} params.options List of options to be rendered
+ * @param {string[]} params.value List of selected values
+ * @param {string} params.parent The parent of these set of options
+ * @param {Function} params.onChange Callback when an option changes
+ */
+const Options = ( {
+	options = [],
+	value = [],
+	parent = '',
+	onChange = () => {},
+} ) => {
+	const [ expanded, setExpanded ] = useState( [] );
+	/**
+	 * Returns true if all the children for the parent are selected
+	 *
+	 * @param {Option} option The parent option to check
+	 */
+	const isParentSelected = ( option ) => {
+		if ( ! option.children ) {
+			return false;
+		}
+
+		return option.children.every(
+			( child ) =>
+				value.includes( child.value ) || isParentSelected( child )
+		);
+	};
+
+	const toggleExpanded = ( option ) => {
+		setExpanded(
+			expanded.includes( option )
+				? expanded.filter( ( el ) => option !== el )
+				: [ ...expanded, option ]
+		);
+	};
+
+	const isRoot = ( option ) => option.value === '';
```

Comment:
Yes, this could be discussed in the next milestone. @ecgan and I decided to postpone this topic but if it's important for the current implementation we can decide now.

Since it's an array of Values I don't think there is an issue with that...  the expected "None" value will be `[]` in this case instead of ''. However, we can have some default value other than `''` based on that neither you nor Gan seems to like it.  Maybe `{}` as suggested or just something more complex like `"__ROOT__" `

Also, we discussed having the prop  `selectAllOption` that allows the user to override completely the root, not only the label.

Reply:
I noticed that in PR #1294 it was decided and recorded to Milestone 3a. However, when reading the code, it looks like a bug because the JSDoc of `Option.value` says its type is a string.

Even though known the entire "Select All" implementation may or may not change again, it is obvious that the bug would be noticed, making it possible that every time there is a code review with related changes, the reviewer might need to constantly check the discussions in #1294 and in here, and understand that it's still an open issue.

Since no more complex handling is required to solve the problem in the current implementation, perhaps it is still worthwhile to make some adjustments before progressing to Milestone 3a, at least one of these is recommended.
- Add code comments to **index.js** and **options.js** to indicate this's a known issue and will be solved in Milestone 3a.
- Adjust the code to make it logically correct from the programming perspective.